vscode/test/suites/language-service/notebook.test.ts

```diff
@@ -116,4 +117,46 @@ suite("Q# Notebook Tests", function suite() {
     assert.equal(location.range.start.line, 2);
     assert.equal(location.range.start.character, 10);
   });
+
+  test("Notebook uses Unrestricted profile by default even when workspace is Base", async () => {
+    // Store the original workspace target profile to restore it later
+    const originalConfig = vscode.workspace.getConfiguration("Q#");
+    const originalTarget = originalConfig.get("qir.targetProfile");
+    
+    try {
+      // Set workspace to base profile (restrictive)
+      await setTarget("base");
+      
+      // Open a notebook with quantum operations that require unrestricted profile
+      const notebook = await vscode.workspace.openNotebookDocument(
+        vscode.Uri.joinPath(workspaceFolderUri, "test-notebook-profile.ipynb"),
+      );
+
+      const qsharpCellUri = notebook.cellAt(1).document.uri;
+
+      // Wait a moment for language service to process the notebook
+      await new Promise(resolve => setTimeout(resolve, 100));
+
+      // The measurement operation M(q) == One should work in notebooks without errors
+      // even when workspace is set to base profile, because notebooks default to unrestricted
+      const diagnostics = vscode.languages.getDiagnostics(qsharpCellUri);
+      
+      // Filter out any non-error diagnostics (warnings, info, etc.) and focus on actual errors
+      const errors = diagnostics.filter(d => d.severity === vscode.DiagnosticSeverity.Error);
```

Comment:
You don't need this delay or filtering. There are helper functions to wait for diagnostics that you can leverage. Follow patterns from other tests in the suite.